### ISSUE TITLE
Parser.parse adds "missing" to original_data

### ISSUE THREAD

**Issue by lafrech** [marshmallow-code/webargs#146]
When parsing input, parser.parse adds `'field': missing` to `original_data` for every missing field. This happens only at first level, not in nested schema.

Hopefully, this test case will make this clearer:

```py
from marshmallow import Schema, fields, validates_schema

from webargs.flaskparser import parser

class Location(Schema):

    city = fields.Str()
    country = fields.Str()

class User(Schema):

    name = fields.Str()
    location = fields.Nested(Location)

    @validates_schema(pass_original=True)
    def print_original_data(self, data, original_data):
        print(original_data)

    class Meta:
        strict = True

input_1 = {'name': 'Eric Cartman'}
input_2 = {'location': {'city': 'South Park'}}

# Via pure Marshmallow
User().load(input_1)
User().load(input_2)
# {'name': 'Eric Cartman'}
# {'location': {'city': 'South Park'}}

# Mock Flask request object
class Request():
    def __init__(self, args):
        self.args = args

req_1 = Request(input_1)
req_2 = Request(input_2)

# Via webargs
parser.parse(User, req_1, locations=['query'])
parser.parse(User, req_2, locations=['query'])
# {'location': <marshmallow.missing>, 'name': 'Eric Cartman'}
# {'location': {'city': 'South Park'}, 'name': <marshmallow.missing>}
```

AFAIKT, this should not happen. I'd expect original_data to be the same as when using pure Marshmallow.

I could fix this by modifying `_parse_request`like this:

```py
            argdict = schema.fields
            parsed = {}
            for argname, field_obj in iteritems(argdict):
                parsed_value = self.parse_arg(argname, field_obj, req, locations)
                # If load_from is specified on the field, try to parse from that key
                if parsed_value is missing and field_obj.load_from:
                    parsed_value = self.parse_arg(field_obj.load_from, field_obj, req, locations)
                    argname = field_obj.load_from
                # Only add value if not missing
                if parsed_value is not missing:
                    parsed[argname] = parsed_value
```

Is this the correct thing to do?

**Comment by lafrech** [comment]
FWIW, here's my fix branch: https://github.com/sloria/webargs/compare/dev...Nobatek:fix_original_data?expand=1

**Comment by sloria** [comment]
@lafrech Thank you for reporting this. This is a legitimate bug and I was able to reproduce it (thanks for the example).

I've created a PR with your fix in #147
